packages/core/src/node/env-variables/env-variables-server.ts

```diff
@@ -14,10 +14,18 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
+import * as path from 'path';
+import * as os from 'os';
 import { injectable } from 'inversify';
 import { EnvVariable, EnvVariablesServer } from '../../common/env-variables';
 import { isWindows } from '../../common/os';
 
+const THEIA_DATA_FOLDER = '.theia';
```

Comment:
Instead of writing `.theia` anywhere, it would be nice to derive it from `FrontendApplicationConfig#applicationName`. So if one overrides the default Theia app name, it gets its own data folder without any further customization.

Reply:
I may do that, but `applicationName` is more like a title, whereas here we need data folder name (`Theia` vs `.theia`).

Reply:
Would this work?
```js
`.${applicationName.toLocaleLowerCase()}`
```

Reply:
As for a generic approach, we can mimic [`_.camelCase`](https://lodash.com/docs/4.17.15#camelCase).

Reply:
Electron uses `name` for id and `productName` for title: https://www.electron.build/configuration/configuration#configuration It would be the best to align. You can use `ApplicationPackage` class to access `name` property of package.json if it is missing then falling back to `.theia` is fine fro backward compatibility. Unfortunate we cannot rename `applicationName` to `packageName` anymore.

Reply:
Actually, please leave it `.theia` by default, products have to override `EnvVariableServer`. It is fine, since DI is primarily way to customize. Changing it to something else will break all downstream projects which wants to use `.theia` but provide `applicationName` or have another `name` in package.json.

Reply:
Not sure that this is a good idea. We are trying to guess data folder name from title, which is fragile and might no behave as expected.

Reply:
@mmorhun it is good, like it now. We cannot change it to something else. It will break downstream projects. Projects which want to use another folder have to override.